code/game/machinery/doors/airlock.dm

```diff
@@ -1161,20 +1161,31 @@ GLOBAL_LIST_EMPTY(airlock_emissive_underlays)
 /obj/machinery/door/airlock/try_to_crowbar(mob/living/user, obj/item/I)
 	if(operating)
 		return
-	if(istype(I, /obj/item/twohanded/fireaxe)) //let's make this more specific //FUCK YOU
-		var/obj/item/twohanded/fireaxe/F = I
-		if(F.wielded)
-			if(density && !prying_so_hard)
-				playsound(src, 'sound/machines/airlock_alien_prying.ogg', 100, 1) //is it aliens or just the CE being a dick?
-				prying_so_hard = TRUE //so you dont pry the door when you are already trying to pry it
-				var/result = do_after(user, 5 SECONDS, target = src)
-				prying_so_hard = FALSE
-				if(result)
-					open(TRUE)
-					if(density && !open(TRUE))
-						to_chat(user, "<span class='warning'>Despite your attempts, [src] refuses to open.</span>")
-		else
-			to_chat(user, "<span class='warning'>You need to be wielding the fire axe to do that!</span>")
+	if(HAS_TRAIT(I, TRAIT_FORCES_OPEN_DOORS_ITEM))
+		/// Used with an istype check to find out if it's a wielded item or not
+		var/obj/item/twohanded/twohanded_item = I
+		/// Time it takes to open an airlock with an item with the TRAIT_FORCES_OPEN_DOORS_ITEM trait, 5 seconds for wielded items, 10 seconds for nonwielded items
+		var/time_to_open_airlock = 10 SECONDS
+		/// Can we open the airlock while unpowered? Wielded item's can't, but unwielded items can
+		var/can_force_open_while_unpowered = TRUE
+		if(istype(twohanded_item))
+			can_force_open_while_unpowered = FALSE
+			time_to_open_airlock = 5 SECONDS
+			if(!twohanded_item.wielded)
+				to_chat(user, "<span class='warning'>You need to be wielding [I] to do that!</span>")
+				return
+		if(density && !prying_so_hard)
+			playsound(src, 'sound/machines/airlock_alien_prying.ogg', 100, 1) //is it aliens or just the CE being a dick?
+			if(!arePowerSystemsOn() && can_force_open_while_unpowered)
+				open(2)
+				return
+			prying_so_hard = TRUE //so you dont pry the door when you are already trying to pry it
+			var/result = do_after(user, time_to_open_airlock, target = src)
+			prying_so_hard = FALSE
+			if(result)
+				open(TRUE)
+				if(density && !open(TRUE))
+					to_chat(user, "<span class='warning'>Despite your attempts, [src] refuses to open.</span>")
```

Comment:
Can you make this into a proc? To keep the current proc a bit cleaner

Reply:
done